pandas/tests/frame/constructors/test_from_records.py

```diff
@@ -184,7 +187,10 @@ def test_from_records_bad_index_column(self):
         tm.assert_index_equal(df1.index, Index(df.C))
 
         # should fail
-        msg = r"Shape of passed values is \(10, 3\), indices imply \(1, 3\)"
+        msg = (
```

Comment:
can you write this as `"|".join([...])`

Reply:
I think we have plenty of places where we use this pattern as well? (I don't know if there is a "preferred" way)

Reply:
we do use both patterns, but ive been trying to move towards the `"|".join` one, in large part bc i occasionally see typos like `msg = "foo|bar|"` (trailing pipe) that make the regex useless

Reply:
OK, updated to use `"|".join`. I think a slight disadvantage of that is it needs one additional level of indentation (because of the list inside `join`), but that's only minor ;)

Reply:
thanks